### PR TITLE
Debug messages in fork GC

### PR DESCRIPTION
**Pull request by alonre24**
**Describe the changes in the pull request**

For debugging MOD-6640 where we see that the child process is stuck without any clear reason - add debug messages in child and parent handlers so we can have more information in the future. 

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/fork_gc.c [hunk 1]
```diff
@@ -513,12 +513,12 @@ static void FGC_childCollectTags(ForkGC *gc, RedisSearchCtx *sctx) {
 }
 
 static void FGC_childScanIndexes(ForkGC *gc, IndexSpec *spec) {
-  RedisSearchCtx sctx = SEARCH_CTX_STATIC(gc->ctx, spec);
-
-  FGC_childCollectTerms(gc, &sctx);
-  FGC_childCollectNumeric(gc, &sctx);
-  FGC_childCollectTags(gc, &sctx);
-
+	RedisSearchCtx sctx = SEARCH_CTX_STATIC(gc->ctx, spec);
```
Comment (GuyAv46) [inline src/fork_gc.c:516]: let's keep the indentations to 2 spaces per level

File: src/fork_gc.c [hunk 2]
```diff
@@ -513,12 +513,12 @@ static void FGC_childCollectTags(ForkGC *gc, RedisSearchCtx *sctx) {
...
+	RedisModule_Log(sctx.redisCtx, "debug", "ForkGC in index %s - child scanning indexes start", sctx.spec->name);
```
Comment (GuyAv46) [inline src/fork_gc.c:517]: Where will we see the fork logs?

**Review by GuyAv46 — COMMENTED** [review]
Consider logging before and after the vecsim GC as well (or inside the function only if there are vecsim fields)

File: src/fork_gc.c (hunk 1 above)
Comment (alonre24) [reply src/fork_gc.c:516]: ok

File: src/fork_gc.c (hunk 2 above)
Comment (alonre24) [reply src/fork_gc.c:517]: in the same log file

**Comment by alonre24** [comment]
> Consider logging before and after the vecsim GC as well (or inside the function only if there are vecsim fields)

we have it (in vecsim)

File: src/fork_gc.c [hunk 3]
```diff
@@ -86,8 +86,8 @@ static int __attribute__((warn_unused_result)) FGC_recvFixed(ForkGC *fgc, void *
       buf += nrecvd;
       len -= nrecvd;
     } else if (nrecvd < 0 && errno != EINTR) {
-      printf("Got error while reading from pipe (%s)", strerror(errno));
-      return REDISMODULE_ERR;
+	  RedisModule_Log(fgc->ctx, "verbose", "ForkGC - got error while reading from pipe (%s)", strerror(errno));
+	  return REDISMODULE_ERR;
```
Comment (GuyAv46) [inline src/fork_gc.c:90]: The indentation here is still with tabs
```suggestion
    RedisModule_Log(fgc->ctx, "verbose", "ForkGC - got error while reading from pipe (%s)", strerror(errno));
    return REDISMODULE_ERR;
```

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**